rest/user_api_test.go

```diff
@@ -1507,3 +1507,68 @@ func TestUnauthorizedAccessForDB(t *testing.T) {
 	require.Contains(t, response.Body.String(), ErrInvalidLogin.Message)
 
 }
+
+func TestUserMultipleDBs(t *testing.T) {
+	base.RequireNumTestDataStores(t, 2)
+	rt := NewRestTester(t, &RestTesterConfig{
+		PersistentConfig: true,
+	})
+	defer rt.Close()
+
+	defaultCollectionDB := "defaultcollectiondb"
+	namedCollectionDB := "namedcollectiondb"
+	// create database using _default._default
+	dbConfig := rt.NewDbConfig()
+	dbConfig.Scopes = nil
+	RequireStatus(t, rt.CreateDatabase(defaultCollectionDB, dbConfig), http.StatusCreated)
+
+	// create a database with a named collection
+	dbConfig = rt.NewDbConfig()
+	dbConfig.Scopes = GetCollectionsConfig(t, rt.TestBucket, 1)
+	RequireStatus(t, rt.CreateDatabase(namedCollectionDB, dbConfig), http.StatusCreated)
+
+	// create matching named user and roles in both databases
+	for _, dbName := range []string{defaultCollectionDB, namedCollectionDB} {
```

Comment:
I feel like it would be useful to have more interleaved user IDs (alphabetically), to make sure there isn't anything unexpected going on related to ordering.  e.g.
_sync:user:alice
_sync:user:namedcollectiondb:alice
_sync:user:nancy
etc.